src/lib/utils/index.ts

```diff
@@ -19,22 +27,38 @@ const uploadApiEndpoint = `https://zns-backend.netlify.app/.netlify/functions/up
  * @param params Metadata parameters
  * @returns URI to the created Metadata
  */
-
-export const createDomainMetadata = async (params: DomainMetadataParams) => {
-	// upload image to http backend
-
+const uploadMetadata = async (params: DomainMetadataParams) => {
+	// upload images to http backend
 	const imageResponse = await fetch(uploadApiEndpoint, {
 		method: 'POST',
 		body: JSON.stringify(params.image),
 	});
 	const image = (await imageResponse.json()) as UploadResponseDTO;
 
-	// upload metadata to IPFS
-	const metadataObject = {
+	let metadataObject: MetadataObject = {
 		name: params.name,
 		description: params.story,
-		image: image.url,
+		imageUri: image.url,
 	};
+
+	if (params.previewImage) {
+		//if params has the optional preview image it builds metadata with it
+		const previewImageResponse = await fetch(uploadApiEndpoint, {
+			method: 'POST',
+			body: JSON.stringify(params.previewImage),
+		});
+		const previewImage =
+			(await previewImageResponse.json()) as UploadResponseDTO;
+		metadataObject.previewImageUri = previewImage.url;
+		return metadataObject;
```

Comment:
Is this return statement needed? Or can it be removed and the code will function exactly the same?